### PR TITLE
More customizable preview

### PR DESCRIPTION
**Pull request by kristofferj**
This is just for TESTING.
Please provide feedback. Can split into multiple branches/PR if wanted.

* Previews take the props title, subtitle, media, description
* New Color-input plugin
* These props can me string, element/object/component or function
* sanityDefaultPreview handles legacy props like image and imageUrl
* Documentspane with card and media (thumbs) preview is enhanced and available for developers (__DEV__)
* Fixes bugs created in the focus makover
* Fixes margins in fieldset

### CODE REVIEW

File: packages/@sanity/color-input/sanity.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "paths": {
+    "source": "./src",
+    "compiled": "./lib"
+  },
+  "parts": [
```
Comment (rexxars) [inline packages/@sanity/color-input/sanity.json]: Not sure we need to expose these parts. Instead, we should implement the schema type part.
See https://github.com/sanity-io/plugin-template-chess-input/blob/master/template/sanity.json for an example

File: packages/@sanity/color-input/src/schema.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      title: 'Hex',
+      name: 'hex',
+      type: 'string'
+    },
+    {
+      title: 'Alpha',
+      name: 'alpha',
+      type: 'number'
+    }
+  ],
```
Comment (rexxars) [inline packages/@sanity/color-input/src/schema.js]: `hsl` seems to be missing.

File: packages/@sanity/color-input/src/schema.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        subtitle: subtitle,
+        media: () => {
+          return (
+            <div
+              style={{
+                backgroundColor: hex || '#000',
+                opacity: alpha || 1,
+                position: 'absolute', height: '100%', width: '100%', top: '0', left: '0'
+              }}
+            />
```
Comment (rexxars) [inline packages/@sanity/color-input/src/schema.js]: Could be simplified to `media: () => (<div... />)`

File: packages/test-studio/package.json [hunk 4]
```diff
@@ -22,6 +22,7 @@
     "@sanity/base": "^0.123.0",
     "@sanity/cli": "^0.123.0",
     "@sanity/code-input": "^0.123.0",
+    "@sanity/color-input": "^0.122.2",
```
Comment (rexxars) [inline packages/test-studio/package.json]: Remember to bump this to 0.123.0 as well

File: packages/test-studio/schemas/schema.js [hunk 5]
```diff
@@ -60,6 +65,7 @@ export default createSchema({
     recursiveArray,
     myObject,
     codeInputType,
+    colorInputType,
```
Comment (rexxars) [inline packages/test-studio/schemas/schema.js]: This shouldn't be necessary when the color input implements the schema type part

File: packages/@sanity/color-input/src/RenderColor.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import React, {PureComponent} from 'react'
+
+export default function(props) {
+  console.log(this.props)
```
[the remark this answers is not in the document]
Comment (kristofferj) [reply packages/@sanity/color-input/src/RenderColor.js]: Nope. removing

File: packages/@sanity/color-input/src/schema.js (hunk 2 above)
Comment (kristofferj) [reply packages/@sanity/color-input/src/schema.js]: `hsl` is optional. Its discussed with @simen

File: packages/@sanity/color-input/sanity.json (hunk 1 above)
Comment (kristofferj) [reply packages/@sanity/color-input/sanity.json]: 👍

File: packages/@sanity/color-input/src/ColorInput.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
+import React, {PureComponent} from 'react'
+import PropTypes from 'prop-types'
+import PatchEvent, {set, merge} from '@sanity/form-builder/PatchEvent'
```
[the remark this answers is not in the document]
Comment (kristofferj) [reply packages/@sanity/color-input/src/ColorInput.js]: 👍

File: packages/@sanity/color-input/src/ColorInput.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    const {onChange} = this.props
+
+    if (!color) {
+      console.error('Color missing') //eslint-disable-line
+    }
+
+    color._type = 'color'
+    color.alpha = color.rgb.a
+
+    onChange(PatchEvent.from(
```
[the remark this answers is not in the document]
Comment (kristofferj) [reply packages/@sanity/color-input/src/ColorInput.js]: This code was written with @simen. Can look into not mutate directly

[the remark this answers is not in the document]
Comment (kristofferj) [reply packages/@sanity/color-input/src/ColorInput.js]: 👍

File: packages/@sanity/color-input/src/schema.js (hunk 3 above)
Comment (kristofferj) [reply packages/@sanity/color-input/src/schema.js]: 👍

**Pull request merged**